gitfile/tokenfetcher.go

```diff
@@ -21,13 +23,13 @@ type HeaderStruct struct {
 // TokenFetcher is the interface for the authentication token suppliers which are provides tokens as a HeaderStruct
 // instances
 type TokenFetcher interface {
-	BuildHeader(repoUrl string) HeaderStruct
+	BuildHeader(ctx context.Context, repoUrl string, loginCallback func(url string)) (*HeaderStruct, error)
```

Comment:
If the `SPITokenFetcher` held the callback as its field (as suggested above), you wouldn't have to pass the callback down the call hierarchy.